tests/unittest/aws_test.py

```diff
@@ -322,16 +338,47 @@ def testCreateVolumeFromSnapshot(self, mock_ec2_api):
     self.assertEqual(
         'prefix-fake-snapshot-d69d57c3-copy', volume_from_snapshot.name)
 
+  @mock.patch('libcloudforensics.aws.AWSAccount._ReadStartupScript')
+  @mock.patch('libcloudforensics.aws.AWSAccount.GetInstancesByName')
+  @mock.patch('libcloudforensics.aws.AWSAccount.ClientApi')
+  def testGetOrCreateAnalysisVm(self,
+                                mock_ec2_api,
+                                mock_get_instance,
+                                mock_script):
+    """Test that a VM is created or retrieved if it already exists."""
+    mock_ec2_api.return_value.run_instances.return_value = MOCK_RUN_INSTANCES
+    mock_ec2_api.return_value.get_waiter.return_value.wait.return_value = None
+    mock_get_instance.return_value = [FAKE_INSTANCE_WITH_NAME]
+    mock_script.return_value = ''
+
+    # GetOrCreateAnalysisVm(vm_name, boot_volume_size, AMI, cpu_cores) where
+    # vm_name is the name of an analysis instance that already exists.
+    vm, created = FAKE_AWS_ACCOUNT.GetOrCreateAnalysisVm(
+        FAKE_INSTANCE_WITH_NAME.name, 1, 'ami-id', 2)
+    self.assertIsInstance(vm, aws.AWSInstance)
+    self.assertEqual('fake-instance', vm.name)
+    self.assertFalse(created)
+
+    # GetOrCreateAnalysisVm(non_existing_vm, boot_volume_size, AMI, cpu_cores).
+    # We mock the GetInstanceById() call to throw a RuntimeError to mimic
+    # an instance that wasn't found.
+    mock_get_instance.side_effect = RuntimeError()
+    vm, created = FAKE_AWS_ACCOUNT.GetOrCreateAnalysisVm(
+        'non-existent-instance-name', 1, 'ami-id', 2)
+    self.assertIsInstance(vm, aws.AWSInstance)
+    self.assertEqual('non-existent-instance-name', vm.name)
+    self.assertTrue(created)
```

Comment:
The behavior this is testing is identical to the behavior in the previous block. Which actually kinda matches what I expect - GetOrCreateAnalysisVm will create the analysis VM if it doesn't exist.

Isn't there a "create" function you can mock and assert it's actually being called in this case and not in the previous one? (see my comment below)

Reply:
I updated that with an additional assertion on `mock_ec2_api.return_value.run_instances.assert_not_called()` in the first block, and `mock_ec2_api.return_value.run_instances.assert_called()` in the second block. The `run_instances` call is the boto3 method to create instances.